cosmoz-grouped-list.js

```diff
@@ -99,20 +100,93 @@
 		 */
 		_renderedItems: null,
 
+		/**
+		 * Instances of Template that are NOT available for reuse at the moment.
+		 */
+		_usedInstances: [],
+
+		/**
+		 *  Instances of Template that are available for reuse at the moment.
+		 */
+		_reusableInstances: [],
+
+		/**
+		 * Polymer `created` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		created() {
 			this._templateSelectors = [];
 			this._renderedItems = [];
-			this._physicalItens = [];
 		},
 
+		/**
+		 * Polymer `attached` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		attached() {
-			this._isAttached = true;
+			this._templatesObserver = Polymer.dom(this.$.templates)
+				.observeNodes(this._onTemplatesChange.bind(this));
 			this._debounceRender();
 		},
 
+		/**
+		 * Polymer `detached` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		detached() {
-			this._isAttached = false;
+			if (this._templatesObserver) {
+				Polymer.dom(this).unobserveNodes(this._templatesObserver);
+				this._templatesObserver = null;
+			}
 			this.cancelDebouncer('render');
+
+
+			this._resetAllTemplates();
+
+			this._usedInstances = [];
+			this._reusableInstances = [];
+
+		},
+		_onTemplatesChange(change) {
```

Comment:
I don't think we have to observe for templates change. 
Also, the code below does not seem to really handle template changes, since it won't re-render the list if  `_ctors` is not empty.
I think we should keep it simple, and just use templatize on first render, without using an observer for the templates.

Reply:
OK then let's keep the node observer